MergerLogic/Utils/HttpRequestUtils.cs

```diff
@@ -0,0 +1,99 @@
+namespace MergerLogic.Utils
+{
+    public class HttpRequestUtils : IHttpRequestUtils
+    {
+        private HttpClient _httpClient;
+
+        public HttpRequestUtils(HttpClient httpClient)
+        {
+            this._httpClient = httpClient;
+        }
+
+        ~HttpRequestUtils()
+        {
+            if (this._httpClient != null)
+            {
+                this._httpClient.Dispose();
+            }
+        }
+
+        private HttpContent? GetContent(HttpRequestMessage req)
+        {
+            var resTask = this._httpClient.SendAsync(req);
+            resTask.Wait();
+
+            var httpRes = resTask.Result;
+            if (httpRes.StatusCode == System.Net.HttpStatusCode.NotFound)
+            {
+                return null;
```

Comment:
should be handled on higher level (either throw - more conventional, or receive param that it should be ignored - more efficient)